core/src/rules/deindent_package_directives.rs

```diff
@@ -0,0 +1,168 @@
+use crate::prelude::*;
+
+pub struct DeindentPackageDirectives {}
+
+impl LogicalLinesConsolidator for DeindentPackageDirectives {
+    fn consolidate(&self, (tokens, lines): (&mut [Token], &mut [LogicalLine])) {
+        match tokens
+            .iter()
+            .map(TokenData::get_token_type)
+            .find(|tt| !tt.is_comment_or_directive())
+        {
+            Some(TokenType::Keyword(KeywordKind::Package)) => {}
+            _ => return,
+        }
```

Comment:
I think something like this makes a little more sense. The random brackets make rustfmt put it all on the same line.
```suggestion
        let Some(TokenType::Keyword(KeywordKind::Package)) =
            (tokens.iter().map(TokenData::get_token_type)).find(|tt| !tt.is_comment_or_directive())
        else {
            return;
        };
```